crates/ra_hir_ty/src/traits/chalk/tls.rs

```diff
@@ -113,14 +134,15 @@ impl DebugContext<'_> {
         };
         let trait_data = self.0.trait_data(trait_);
         let params = alias.substitution.parameters(&Interner);
-        write!(
-            fmt,
-            "<{:?} as {}<{:?}>>::{}",
-            &params[0],
-            trait_data.name,
-            &params[1..],
-            type_alias_data.name
-        )
+        write!(fmt, "<{:?} as {}", &params[0], trait_data.name,)?;
+        if params.len() > 1 {
+            write!(
+                fmt,
+                "<{}>",
+                &params[1..].iter().format_with(", ", |x, f| f(&format_args!("{:?}", x))),
```

Comment:
TIL that `format` and `format_with` are a thing. I wonder if that means that we should kill `stdx::SepBy`...

Reply:
Yeah, I also just found them for this... don't actually remember how, maybe Chalk used them :thinking: